Libs/MRML/Core/vtkMRMLTableStorageNode.cxx

```diff
@@ -340,106 +340,273 @@ bool vtkMRMLTableStorageNode::ReadTable(std::string filename, vtkMRMLTableNode*
     return 0;
     }
 
-  vtkSmartPointer<vtkTable> table = vtkSmartPointer<vtkTable>::New();
-  for (int col = 0; col < rawTable->GetNumberOfColumns(); ++col)
+  // Struct for managing columns
+  typedef struct
+  {
+    std::string ColumnName = "";
+    std::vector<vtkAbstractArray*> RawComponentArrays;
+    int DataType = -1;
```

Comment:
Maybe ScalarType or ValueType could be more clear (for me, "data type" definition includes everything - that it is a scalar/vector, with N components)